src/dotnet-interactive-vscode/common/ipynbUtilities.ts

```diff
@@ -127,20 +127,29 @@ export const requiredLanguageInfoData = {
 };
 
 export function withDotNetKernelMetadata(metadata: { [key: string]: any } | undefined): any | undefined {
-    const result = {
+    let result = {
         ...metadata,
         custom: {
             ...metadata?.custom,
             metadata: {
                 ...metadata?.custom?.metadata,
                 kernelspec: {
-                    ...metadata?.custom?.metadata?.kernelspec,
-                    ...requiredKernelspecData,
-                },
-                language_info: requiredLanguageInfoData,
+                    ...metadata?.custom?.metadata?.kernelspec
+                }
             },
         }
     };
+    let name = result?.custom?.metadata?.kernelspec?.name
+
+    if (typeof name !== 'string' || !name.startsWith(".net-")) {
+        result.custom.metadata.kernelspec = {
+            ...metadata?.custom?.metadata?.kernelspec,
+            ...requiredKernelspecData,
+        };
+
+        result.custom.metadata.language_info = { ...requiredLanguageInfoData };
```

Comment:
This isn't quite correct, because `requiredLanguageInfoData` always defaults to C#.  I think the best solution would be to leave the metadata untouched if the kernelspec name was determined to be good, e.g.,

``` typescript
const kernelspecName = metadata?.custom?.metadata?.kernelspec?.name;
if (typeof kernelspecName === 'string' && kernelspecName.startsWith('.net-')) {
    return metadata; // don't change anything
}

// otherwise the previous contents of this method as they were before
```